src/aurelia-slickgrid/services/aureliaUtil.service.ts

```diff
@@ -1,73 +1,22 @@
-import {
-  inject,
-  Container,
-  createOverrideContext,
-  singleton,
-  ViewCompiler,
-  ViewResources,
-  ViewSlot,
-} from 'aurelia-framework';
-import { AureliaViewOutput } from '../models/index';
+import { AureliaViewOutput, ViewModelBindableInputData } from '../models/index';
+import { Constructable, CustomElement, IAurelia, singleton } from 'aurelia';
 
-@singleton(true)
-@inject(
-  Container,
-  ViewCompiler,
-  ViewResources
-)
+@singleton()
 export class AureliaUtilService {
   constructor(
-    private readonly container: Container,
-    private readonly viewCompiler: ViewCompiler,
-    private readonly viewResources: ViewResources,
+    @IAurelia private readonly au: IAurelia
   ) { }
 
-  createAureliaViewModelAddToSlot(templateUrl: string, bindableData: any, targetElement?: HTMLElement | Element, clearTargetContent = false): AureliaViewOutput | null {
-    const viewFactory = this.viewCompiler.compile('<template><compose view-model.bind="template" view-model.ref="viewModelRef"></compose></template>', this.viewResources);
-
+  async createAureliaViewModelAddToSlot(viewModel: Constructable, bindableData: ViewModelBindableInputData, targetElement?: HTMLElement | Element, clearTargetContent = false): Promise<AureliaViewOutput | null> {
     if (targetElement) {
+      // TODO: MB - is this needed?
       if (clearTargetContent && targetElement.innerHTML) {
```

Comment:
yeah if I remember correctly it removes any html from the RowDetail (or what is behind the opened row detailed) and avoids showing html over html (or row detail on top of slickgrid cells). I think it was for that but I could be wrong

Reply:
I think the question here is whether you can have situations when you DON'T need to clean the inner html?